packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -314,7 +314,10 @@ abstract class RenderViewportBase<ParentDataClass extends ContainerParentDataMix
   void visitChildrenForSemantics(RenderObjectVisitor visitor) {
     childrenInPaintOrder
         .where(
-          (RenderSliver sliver) => sliver.geometry!.visible || sliver.geometry!.cacheExtent > 0.0,
+          (RenderSliver sliver) =>
+              sliver.geometry!.visible ||
+              sliver.geometry!.cacheExtent > 0.0 ||
```

Comment:
looks like there as long as the sliver has cacheExtent they will also be preserved. can the SliverEnsureSemantics sets a cacheExtent in itself to avoid adding ensureSemantics?

Reply:
I think relying on the slivers `cacheExtent` wouldn't work for the `describeApproximatePaintClip` and `describeSemanticsClip` case either since we wouldn't be able to differentiate between an item that is invisible but does not want to be clipped vs an item that is invisible but should be clipped, maybe if we use a magic number but not sure how stable that would be (would a normal sliver ever reach this magic number).

Reply:
agree, if it has other implication then we better off implement a new api for this